rusty_pizza_server/src/order_model/order.rs

```diff
@@ -653,4 +668,63 @@ mod tests {
         //Then
         assert_eq!(Err(expected_change), calculated_change);
     }
+
+    #[rstest]
+    #[case(Money::new(5, 50), Money::new(7, 42), 0, Some(Money::new(5, 50)))]
+    #[case(Money::new(2, 34), Money::new(6, 78), 1, Some(Money::new(6, 78)))]
+    #[case(Money::new(3, 21), Money::new(9, 99), 2, None)]
+    fn correct_total_price_for_user_is_returned(
+        #[case] manager_price: Money,
+        #[case] user_price: Money,
+        #[case] asked_id: u32,
+        #[case] expected_total: Option<Money>,
+    ) {
+        // Given:
+        let manager_id = 0;
+        let mut order = Order::new(manager_id);
+
+        let user_id = 1;
+        order.add_user(user_id);
+
+        //When
+        order
+            .add_meal_for_user(0, String::from("0"), String::from("Gross"), manager_price)
+            .unwrap();
+        order
+            .add_meal_for_user(1, String::from("1"), String::from("Tortellini"), user_price)
+            .unwrap();
+
+        //Then
+        assert_eq!(expected_total, order.user_has_to_pay(asked_id));
```

Comment:
You're testing the `user_has_to_pay` function in this test so that should go into the "when" part (not the assertion of course). What is now in the "when" part belongs into the "given" part.

Reply:
done